### PR TITLE
[RFC][UN-6335] Create the components on uniq framework

### PR DESCRIPTION
**Pull request by mmelo**
Refers to: https://uniplaces.atlassian.net/browse/UN-6335

Created uniq v2.0.0 - following the new design language for both smartphone and desktop.

### CODE REVIEW

File: src/images/flag.svg [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+<use xlink:href="#quina" y="-26.016"/>

+<use xlink:href="#quina" x="-20.799"/>

+<use xlink:href="#quina" x="20.745"/>

+<use xlink:href="#quina" y="25.784"/>

+</g>
+</svg>
```
Comment (niceoutput) [inline src/images/flag.svg]: eol

File: src/index.html [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+  </header>
+  <div class="progress-bar">
+    <div class="progress-bar__bar" style="width: 14%"></div>
+  </div>
+  
+  <section id="main" class="main">
+    <div class="main__wrapper">
+      <div class="main__content">
+        <div class="title">How many bedrooms in the property?</div>
+        <div class="input-container ">
```
Comment (niceoutput) [inline src/index.html]: remove white space before closing quotes.

File: src/index.html [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+          <input type="text" class="input input--range" placeholder="2 bedrooms">
+          <div class="input-button input-button--disabled">-</div>
+          <div class="input-button">+</div>
+        </div>
+        <div class="title">How many bathrooms in the property?</div>
+        <div class="input-container">
+          <input type="text" class="input input--range" placeholder="1 bedroom">
+          <div class="input-button input-button--disabled">-</div>
```
Comment (niceoutput) [inline src/index.html]: I have several questions in transforming a div tag in a button, semantically isn't correct...
It's just my way of thought, what do you think?
If it was me i'll use an anchor like an <a> or a <button>

File: src/index.html [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+          <div class="input-button">+</div>
```
Comment (niceoutput) [inline src/index.html]: Same as above...

File: src/index.html [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+        </div>
+        
+        <div class="title">Do you have a living room?</div>
+        <div class="button button--secondary">Add living room</div>
```
Comment (niceoutput) [inline src/index.html]: Same as above...

File: src/index.html [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+          <div class="input-button">+</div>
+        </div>
+        <div class="title">How many bathrooms in the property?</div>
+        <div class="input-container">
+          <input type="text" class="input input--range" placeholder="1 bedroom">
+          <div class="input-button input-button--disabled">-</div>
+          <div class="input-button">+</div>
+        </div>
+        
+        <div class="title">Do you have a living room?</div>
```
Comment (niceoutput) [inline src/index.html]: In titles you should use heading tags accordingly the importance.

File: src/index.html [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+        <div class="button button--secondary">Add living room</div>
+        
+        <div class="title">Which amenities does your property have?</div>
+        <div class="input-container input-container--bordered input-container--focused">
+          <div class="checkbox checkbox--checked"></div>
+          <label for="">WiFi</label>
```
Comment (niceoutput) [inline src/index.html]: Label shouldn't have the empty "for" attribute

File: src/index.html [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+        </div>
+        <div class="input-container input-container--bordered">
+          <div class="checkbox"></div>
+          <label for="">Cable TV</label>
```
Comment (niceoutput) [inline src/index.html]: Same as above and all the way down...

File: src/index.html [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+          <div class="radio-button"></div>
+          <label for="">Yes, bottled gas</label>
+        </div>
+        <div class="input-container input-container--bordered">
+          <div class="radio-button"></div>
+          <label for="">No, it's all electric</label>
+        </div>
+        
+        <div class="title">Describe your property</div>
+        <textarea name="" id="" rows="5" class="textarea" placeholder="I.E: Two bedroom apartment with a balcony and great view in trendy Bairro Alto."></textarea>
```
Comment (niceoutput) [inline src/index.html]: Name attribute shouldn't be empty.

File: src/styles/_common/_typography.scss [hunk 10]
```diff
@@ -1,181 +1,42 @@
-@mixin display-4() {
-  font-size: 112px;
-  margin-bottom: $gutter;
-}
+$base-font-size: 18; // Default font-size
```
Comment (niceoutput) [inline src/styles/_common/_typography.scss]: If you're starting a fresh baseline, why not going in REM's?
It's just my opinion :)

File: src/styles/_components/_info-box.scss [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+        position: relative;
+        width: 100%;
+        height: 1px;
+        margin-bottom: 24px;
+        background-color: color('primary', 'divider');
+    }
+    &__content {
+        @include body();
+    }
+}
```
Comment (niceoutput) [inline src/styles/_components/_info-box.scss]: eol

File: src/styles/_components/_inputs.scss [hunk 12]
```diff
@@ -0,0 +1,261 @@
...
+    height: $input-height;
+    border: 1px solid color('borders');
+    font-size: em(24px);
+    font-weight: bold;
+    color: color('primary');
+    
+    &:first-of-type {
+        border-right: 0;
+    }
+    &:last-of-type {
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: needs a line space..

File: src/styles/_components/_inputs.scss [hunk 13]
```diff
@@ -0,0 +1,261 @@
...
+        border-top-right-radius: 4px;
+        border-bottom-right-radius: 4px;
+    }
+    
+    svg {
+        fill: color('primary');
+    }
+    &:hover {
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_inputs.scss [hunk 14]
```diff
@@ -0,0 +1,261 @@
...
+        background-color: color('buttons', 'hover');
+        cursor: pointer;
+    }
+    &:active {
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_inputs.scss [hunk 15]
```diff
@@ -0,0 +1,261 @@
...
+        background-color: color('background', 'light');
+    }
+    &:disabled,
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_inputs.scss [hunk 16]
```diff
@@ -0,0 +1,261 @@
...
+    &--disabled {
+        svg {
+            fill: color('input', 'disabled');
+        }
+    }
+}
+@mixin textarea() {
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_inputs.scss [hunk 17]
```diff
@@ -0,0 +1,261 @@
...
+    height: $checkbox-size;
+    border: 1px solid color('input', 'border');
+    margin-right: 20px;
+    border-radius: 4px;
+    
+    svg {
+        display: none;
+        fill: color('primary');
+    }
+    & + label {
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_inputs.scss [hunk 18]
```diff
@@ -0,0 +1,261 @@
...
+        height: $checkbox-size;
+        line-height: $checkbox-size;
+        font-size: em(20px);
+        color: color('text', 'body');
+        will-change: color;
+    }
+    &:focus,
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_inputs.scss [hunk 19]
```diff
@@ -0,0 +1,261 @@
...
+        border: 1px solid color('primary', 'light');
+        background-color: color('input', 'focus');
+        
+        svg {
+            display: block;
+        }
+        & + label {
+            color: color('primary');
+        }
+        @include breakpoint(small) {
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_inputs.scss [hunk 20]
```diff
@@ -0,0 +1,261 @@
...
+            background-color: color('background');
+        }
+    }
+}
+@mixin radio-button() {
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_inputs.scss [hunk 21]
```diff
@@ -0,0 +1,261 @@
...
+        background-color: color('input', 'border');
+        border-top-left-radius: 4px;
+        border-bottom-left-radius: 4px;
+        font-size: em(20px);
+        color: color('text', 'title');
+
+        &:hover {
+            cursor: pointer;
+        }
+        &:before,
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_inputs.scss [hunk 22]
```diff
@@ -0,0 +1,261 @@
...
+            width: 7px;
+            height: 1px;
+            top: 50%;
+            background-color: color('text', 'title');
+        }
+        &:before {
+            right: 20px;
+            transform: rotate(45deg);
+        }
+        &:after {
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_inputs.scss [hunk 23]
```diff
@@ -0,0 +1,261 @@
...
+            right: 15px;
+            transform: rotate(-45deg);
+        }
+        &--active {
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_inputs.scss [hunk 24]
```diff
@@ -0,0 +1,261 @@
...
+            &:before {
+                transform: rotate(-45deg);
+            }
+            &:after {
+                transform: rotate(45deg);
+            }
+        }
+        &--active &__dropdown {
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_inputs.scss [hunk 25]
```diff
@@ -0,0 +1,261 @@
...
+        }
+        img {
+            display: inline-block;
+            width: 20px;
+            height: 13px;
+            margin-right: 10px;
+            border-radius: 2px;
+            overflow: hidden;
+        }
+        &__dropdown {
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_inputs.scss [hunk 26]
```diff
@@ -0,0 +1,261 @@
...
+                @include align-items-center();
+                height: 46px;
+                line-height: 46px;
+                padding: 20px;
+                
+                &:hover {
+                    cursor: pointer;
+                    background-color: color('input', 'border');
+                }
+                &:last-child {
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_inputs.scss [hunk 27]
```diff
@@ -0,0 +1,261 @@
...
+                    margin-bottom: 0;
+                }
+            }
+        }
+    }
+    input {
```
Comment (niceoutput) [inline src/styles/_components/_inputs.scss]: same as above

File: src/styles/_components/_modals.scss [hunk 28]
```diff
@@ -0,0 +1,52 @@
...
+        background-color: color('background', 'modal');
+        box-shadow: 0 2px 7px 0 rgba(53, 53, 53, 0.5);
+        text-align: center;
+               
+        &__image {
+            width: 170px;
+            height: 160px;
+            margin: 24px auto;
+        }
+        &__title {
```
Comment (niceoutput) [inline src/styles/_components/_modals.scss]: needs a line space

File: src/styles/_components/_modals.scss [hunk 29]
```diff
@@ -0,0 +1,52 @@
...
+            @include subheading();
+            margin-bottom: 8px;
+        }
+        &__message {
```
Comment (niceoutput) [inline src/styles/_components/_modals.scss]: same as above

File: src/styles/_components/_modals.scss [hunk 30]
```diff
@@ -0,0 +1,52 @@
...
+            @include subtitle();
+            color: color('text', 'modal');
+        }
+    }
+}
```
Comment (niceoutput) [inline src/styles/_components/_modals.scss]: eol

File: src/styles/_components/_progress-bar.scss [hunk 31]
```diff
@@ -0,0 +1,33 @@
...
+        position: absolute;
+        display: block;
+        width: 100%;
+        height: 100%;
+        top: 0;
+        left: 0;
+        z-index: 1;
+        background-color: color('primary', 'translucent');
+    }
+    &__bar {
```
Comment (niceoutput) [inline src/styles/_components/_progress-bar.scss]: needs a line space

File: src/styles/_components/_step-card.scss [hunk 32]
```diff
@@ -0,0 +1,77 @@
...
+    min-height: 210px;
+    padding: 32px;
+    margin-right: 20px;
+    background-color: color('background', 'modal');
+    
+    @include breakpoint(small) {
+        width: 100%;
+        min-height: 0;
+        margin-right: 0;
+        //margin-left: 10px;
```
Comment (niceoutput) [inline src/styles/_components/_step-card.scss]: remove commented code

File: src/styles/_components/_step-card.scss [hunk 33]
```diff
@@ -0,0 +1,77 @@
...
+    }
+    &:last-child {
+        margin-right: 0;
+    }
+    &__step {
```
Comment (niceoutput) [inline src/styles/_components/_step-card.scss]: same as above

File: src/styles/_components/_step-card.scss [hunk 34]
```diff
@@ -0,0 +1,77 @@
...
+        width: 100%;
+        margin-bottom: 12px;
+        font-size: em(14px);
+        color: color('text', 'subtitle');
+        text-transform: uppercase;
+    }
+    &__title {
```
Comment (niceoutput) [inline src/styles/_components/_step-card.scss]: same as above

File: src/styles/_components/_step-card.scss [hunk 35]
```diff
@@ -0,0 +1,77 @@
...
+        width: 100%;
+        margin-bottom: 6px;
+        font-size: em(25px);
+        color: color('text', 'title');
+    }
+    &__hint {
```
Comment (niceoutput) [inline src/styles/_components/_step-card.scss]: same as above

File: src/styles/_components/_step-card.scss [hunk 36]
```diff
@@ -0,0 +1,77 @@
...
+        width: 100%;
+        margin-bottom: 14px;
+        font-size: em(14px);
+        color: color('text', 'subheading');
+    }
+    &__progress-bar {
```
Comment (niceoutput) [inline src/styles/_components/_step-card.scss]: same as above

File: src/styles/_components/_step-card.scss [hunk 37]
```diff
@@ -0,0 +1,77 @@
...
+        width: 100%;
+        height: 8px;
+        margin: 15px 0;
+        
+        &-label {
+            font-size: em(18px);
+            color: color('text', 'body');
+        }
+    }
+    &__anchor {
```
Comment (niceoutput) [inline src/styles/_components/_step-card.scss]: same as above

File: src/styles/_components/_step-card.scss [hunk 38]
```diff
@@ -0,0 +1,77 @@
...
+    }
+    &--passed &__title {
+        font-size: em(19px);
+        color: color('text', 'subtitle');
+    }
+    &--passed &__hint,
+    &--passed .button {
+        display: none;
+    }
+}
```
Comment (niceoutput) [inline src/styles/_components/_step-card.scss]: eol

File: src/styles/demo.scss [hunk 39]
```diff
@@ -0,0 +1,170 @@
...
+    @include flex-wrap();
+    position: relative;
+    width: 100%;
+    height: 100%;
+    
+    @include body();
+    font-family: 'Roboto', sans-serif;
+    font-size: 16px;
+}
+.header,
```
Comment (niceoutput) [inline src/styles/demo.scss]: Add line space..

File: src/styles/demo.scss [hunk 40]
```diff
@@ -0,0 +1,170 @@
...
+.main,
+.footer {
```
Comment (niceoutput) [inline src/styles/demo.scss]: Please don't use class names that have the same name of the tag. 
Also a line space as above

File: src/styles/demo.scss [hunk 41]
```diff
@@ -0,0 +1,170 @@
...
+    &__wrapper {
+        @include flex();
+        @include space-between();
+        width: 100%;
+        max-width: 1000px;
+        margin: 0 auto;
+    }
+}
+.header {
```
Comment (niceoutput) [inline src/styles/demo.scss]: same as above

File: src/styles/demo.scss [hunk 42]
```diff
@@ -0,0 +1,170 @@
...
+    @include header();
+    position: fixed;
+    top: 0;
+    left: 0;
+    z-index: 4;
+}
+.main {
```
Comment (niceoutput) [inline src/styles/demo.scss]: Add line space

File: src/styles/demo.scss [hunk 43]
```diff
@@ -0,0 +1,170 @@
...
+    display: table;
+    width: 100%;
+    padding-top: 60px;
+    padding-bottom: 90px;
+    z-index: 1;
+
+    &__wrapper {
+        padding-top: 46px;
+    }
+    &__content {
```
Comment (niceoutput) [inline src/styles/demo.scss]: Add line space.

File: src/styles/demo.scss [hunk 44]
```diff
@@ -0,0 +1,170 @@
...
+        width: auto;
+        @include breakpoint(small) {
```
Comment (niceoutput) [inline src/styles/demo.scss]: Same as above

File: src/styles/demo.scss [hunk 45]
```diff
@@ -0,0 +1,170 @@
...
+        border: 0;
+        
+        .button {
+            min-width: 50%;
+            height: 100%;
+            border-radius: 0;
+        }
+    }
+}
+.anchor {
```
Comment (niceoutput) [inline src/styles/demo.scss]: line space

File: src/index.html (hunk 3 above)
Comment (mmelo) [inline src/index.html]: Please disregard structure implemented in the demo as it acts only as a placeholder.

File: src/index.html (hunk 7 above)
Comment (mmelo) [inline src/index.html]: Please disregard structure implemented in the demo as it acts only as a placeholder.

File: src/index.html (hunk 8 above)
Comment (mmelo) [inline src/index.html]: Please disregard structure implemented in the demo as it acts only as a placeholder.

File: src/index.html (hunk 9 above)
Comment (mmelo) [inline src/index.html]: Please disregard structure implemented in the demo as it acts only as a placeholder.

File: src/styles/_components/_footer.scss [hunk 46]
```diff
@@ -0,0 +1,26 @@
...
+        @include space-between();
+        @include space-between();
+        @include limited-full-width();
+        height: 100%;
+    }
+    @include breakpoint(small) {
+        height: 58px;
+        padding: 0;
+    }
+}
```
Comment (mmelo) [inline src/styles/_components/_footer.scss]: done

File: src/styles/_components/_modals.scss (hunk 30 above)
Comment (mmelo) [inline src/styles/_components/_modals.scss]: done

**Comment by mmelo** [comment]
@bo @niceoutput fixed.

**Pull request merged**